File: Raven.Client.Lightweight/Linq/ExpressionInfo.cs
//-----------------------------------------------------------------------
// <copyright file="ExpressionInfo.cs" company="Hibernating Rhinos LTD">
//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------
using System;

namespace Raven.Client.Linq
{
	/// <summary>
	/// This class represents a node in an expression, usually a member - but in the case of dynamic queries the path to a member
	/// </summary>
	public class ExpressionInfo
	{
		/// <summary>
		/// Gets the full path of the member being referred to by this node
		/// </summary>
		public string Path { get; private set; }

		/// <summary>
		/// Gets the actual type being referred to
		/// </summary>
		public Type Type { get; private set; }

		/// <summary>
		/// Whatever the expression is of a nested path
		/// </summary>
		public bool IsNestedPath { get; private set; }

		/// <summary>
		/// Creates an ExpressionMemberInfo
		/// </summary>
		public ExpressionInfo(string path, Type type, bool isNestedPath)
		{
			IsNestedPath = isNestedPath;
			Path = path;
			Type = type;
		}
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/client/linq/ExpressionInfo.java
package net.ravendb.client.linq;

import java.lang.reflect.Field;

/**
 * This class represents a node in an expression, usually a member - but in the case of dynamic queries the path to a member
 */
public class ExpressionInfo {
  private String path;
  private Class<?> clazz;
  private boolean nestedPath;
  private Field maybeProperty;
  public String getPath() {
    return path;
  }
  public void setPath(String path) {
    this.path = path;
  }
  public Class<?> getClazz() {
    return clazz;
  }
  public void setClazz(Class<?> clazz) {
    this.clazz = clazz;
  }
  public boolean isNestedPath() {
    return nestedPath;
  }
  public void setNestedPath(boolean nestedPath) {
    this.nestedPath = nestedPath;
  }
  public Field getMaybeProperty() {
    return maybeProperty;
  }
  public void setMaybeProperty(Field maybeProperty) {
    this.maybeProperty = maybeProperty;
  }

  public ExpressionInfo(String path, Class<?> clazz, boolean nestedPath) {
    super();
    this.path = path;
    this.clazz = clazz;
    this.nestedPath = nestedPath;
  }


}
